### ISSUE TITLE
hooks failed to run because it was already open (on `NixOS`)

### ISSUE THREAD

**Issue by MathisWellmann** [buildkite/agent#2734]
I tried using some plugins (`docker`, `docker-compose`, etc) in my pipeline as such:
```yaml
steps:
  - label: ":docker: :rust: build"
    command: RUSTFLAGS="-D warnings" cargo build
    plugins:
      - docker-compose#v3.0.3:
          run: node
    agents:
      queue: nixos
```
Which is executed by the agent on a `nixos` host, which successfully runs all my other pipelines, but this setup fails to run with the following message:

```shell
$ /var/buildkite/plugins/github-com-buildkite-plugins-docker-compose-buildkite-plugin-v3-0-3/hooks/pre-checkout
🚨 Error: The plugin docker-compose pre-checkout hook failed to run because it was already open. We couldn't find out what process had the hook open
🚨 Error: error running "/tmp/buildkite-agent-bootstrap-hook-runner-1100140013": fork/exec /tmp/buildkite-agent-bootstrap-hook-runner-1100140013: no such file or directory
```

@tomowatt gave insight into why this is happening here: https://github.com/buildkite-plugins/docker-compose-buildkite-plugin/issues/434#issuecomment-2063904376

TLDR: The `/bin/bash` executable exists at a different path in `NixOS` as it departs from the `FHS` (File hierarchy standard) to ensure reproducibility. The plugin scripts rely on `bash` being there and it thus fails.  

The only real issue here is that the above **error message is cryptic** and could be improved.
Fixing the path in the nix package is possible to support existing scripts / plugins.
